packages/react/src/components/LiveKitRoom.tsx

```diff
@@ -78,6 +78,12 @@ export interface LiveKitRoomProps extends Omit<React.HTMLAttributes<HTMLDivEleme
 
   simulateParticipants?: number | undefined;
 
+  /**
+   * Optional side effect to run in parallel with the connection process.
+   * This can be used to enable the microphone preconnect buffer / etc
+   */
+  connectionSideEffect?: (room: Room) => Promise<void>;
+
```

Comment:
This new prop / parameter allows a user to inject in a side effect to run in parallel with the `room.connect(...)` call - an example of its use in `agent-starter-react`:
```tsx
// Within a react component:
const enableMicrophonePreConnectBuffer = useCallback(async (room: Room) => {
  room.localParticipant.setMicrophoneEnabled(true, undefined, {
    preConnectBuffer: appConfig.isPreConnectBufferEnabled,
  });
}, []);

// (later on in the component)
return (
  <LiveKitRoom
    token="..."
    serverUrl="..."
    connectionSideEffect={enableMicrophonePreConnectBuffer}
  >
    {/* rest of application here */}
  </LiveKitRoom>
);
```

Note that I could see `enableMicrophonePreConnectBuffer` being extracted into the eventual agents sdk that has been discussed.